### PR TITLE
NEXUS-8091 Fixes Nexus search in Firefox

### PR DESCRIPTION
**Pull request by djsauble**
Issue: https://issues.sonatype.org/browse/NEXUS-8091

The fix involves a one-line patch to the ExtJS source. I’ve filed a ticket with Sencha, and created a patch override for now. The ticket is at:

https://support.sencha.com/index.php#ticket-21425

### CODE REVIEW

**Comment by kellyrob99** [comment]
This is the closest to a good description I could find of what we seem to be looking at http://stackoverflow.com/questions/1703552/encoding-of-window-location-hash

**Comment by adreghiciu** [comment]
Indeed, the preferred way to add fixes are the overrides, and such we can just drop them when bug is fixed by Sencha.
And open a ticket with Sencha as they usually also provide the fix to put there.

**Comment by djsauble** [comment]
Cool, thanks. I opened a ticket and created a patch override. Review at will.

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js [hunk 1]
```diff
@@ -126,10 +126,7 @@ Ext.define('NX.controller.Bookmarking', {
   onNavigate: function (token) {
     var me = this;
 
-    if (token !== Ext.History.bookmark) {
```
Comment (kellyrob99) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js]: so this looks like it was here to prevent self-navigation - is that no longer an issue? Seems unrelated to the FF bug

**Comment by kellyrob99** [comment]
+1 
@djsauble could we get a link to the Sencha ticket please? Tnx

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js (hunk 1 above)
Comment (djsauble) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js]: Yes, it’s unrelated.

Since onNavigate is triggered when the bookmark changes, can this statement ever be false?

**Comment by djsauble** [comment]
https://support.sencha.com/index.php#ticket-21425

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js (hunk 1 above)
Comment (adreghiciu) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js]: Yes, when you press back/forward

Comment (djsauble) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Bookmarking.js]: k, I’ll revert those changes then.

**Comment by adreghiciu** [comment]
+1

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/Ext/patch/Ticket_21425.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ * https://support.sencha.com/index.php#ticket-21425
+ */
+Ext.define('Ext.patch.Ticket_21425', {
+  override: 'Ext.util.History',
+
+  getHash: function() {
+    // HACK: Firefox decodes the hash when accessed directly, so we need to use
+    // location.href to get it instead
+    // return this.win.location.hash.substr(1);
+    return location.href.split('#').splice(1).join('#');
```
Comment (mcculls) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/Ext/patch/Ticket_21425.js]: any particular reason this is `splice(1)` and not `slice(1)`?

Comment (djsauble) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/Ext/patch/Ticket_21425.js]: No reason. They’re functionally the same, since splice modifies the return from split, and not the original array. Probably slice makes more sense.

**Comment by djsauble** [comment]
Manual merge: https://github.com/sonatype/nexus-oss/commit/1765324e3bb5db056e724aa8bae2e533e2377223
